### PR TITLE
E2vid integration refactor

### PR DESCRIPTION
**Pull request by shiba24**
Still wip.
Modifications on:

- test_e2vid.py -> examples/reconstruction_e2vid.py
- Many linting errors
- Folder rectructure
- Remove unnecessary options, Timers, etc.

Still mypy and test (on CPU) fails. I'm working on it.

### CODE REVIEW

**Comment by shiba24** [comment]
This PR is toward your branch: e2vid_integration. So if I merge this PR your branch will be affected. @friedhelm-h

File: examples/reconstruction_e2vid.py
Comment (friedhelm-h) [inline examples/reconstruction_e2vid.py:1]: I didn't mean to commit this file actually. The reader does not exist. Let's delete it for now and add examples as soon as we have data loaders implemented? We could have 1-2 example files which we use in all examples.

**Comment by friedhelm-h** [comment]
Just for me for clarification and future implementation. I can add `# type: ignore` behind a line to avoid mypy type hinting errors in that line? And how do you decide to do type hinting on a variable like [here](https://github.com/shiba24/event-vision-library/blob/e2vd_integration-refactor/src/evlib/processing/reconstruction/e2vid.py#L62)? Is there a rule?

**Comment by shiba24** [comment]
> I can add # type: ignore behind a line to avoid mypy type hinting errors in that line?

Yes. It is not ideal, but it can be a hotfix -- better refactor in future.

> And how do you decide to do type hinting on a variable like? Is there a rule?

More ad-hoc for me. Sometimes mypy cannot infer the type of the member variable of the class etc. So I'm giving some hints for mypy then it won't complain. The way I fix the lint is quite debuggy.

**Comment by shiba24** [comment]
mypy and tests pass now, I'm fixing the typeguard.

**Pull request merged**